specs/interop/eth-bridging.md

```diff
@@ -0,0 +1,244 @@
+# ETH Bridging
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Overview](#overview)
+- [Design](#design)
+  - [Interface and properties](#interface-and-properties)
+    - [`SuperchainETHBridge`](#superchainethbridge)
+      - [`sendETH`](#sendeth)
+      - [`relayETH`](#relayeth)
+    - [`ETHLiquidity`](#ethliquidity)
+      - [`mint`](#mint)
+      - [`burn`](#burn)
+  - [Events](#events)
+    - [`SuperchainETHBridge`](#superchainethbridge-1)
+      - [`SendETH`](#sendeth)
+      - [`RelayETH`](#relayeth)
+    - [`ETHLiquidity`](#ethliquidity-1)
+      - [`LiquidityMinted`](#liquidityminted)
+      - [`LiquidityBurned`](#liquidityburned)
+- [Constants](#constants)
+- [Invariants](#invariants)
+  - [System level invariants](#system-level-invariants)
+    - [`ETHLiquidity`](#ethliquidity-2)
+  - [Contract level invariants](#contract-level-invariants)
+    - [`SuperchainETHBridge`](#superchainethbridge-2)
+      - [`sendETH`](#sendeth-1)
+      - [`relayETH`](#relayeth-1)
+    - [`ETHLiquidity`](#ethliquidity-3)
+      - [`mint`](#mint-1)
+      - [`burn`](#burn-1)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Overview
```

Comment:
I think it'd be ideal if we actually broke this out into three different documents:

- One document explains the architecture of the ETH bridging system (this document)
- One document clearly specifies the behavior of the `SuperchainETHBridge` contract using the contract specification format that we've been standardizing on. You can find a good example of this specification format [here](https://github.com/ethereum-optimism/specs/blob/main/specs/fault-proof/stage-one/anchor-state-registry.md).
- One document clearly specifies the behavior of the `ETHLiquidity` contract, same standardized format.

Reply:
The main value of the standardized specification format is that it forces you to carefully consider the properties and invariants of each of the contracts. All of the code/behavior of the contract should ultimately serve some invariant that we want to maintain. I think this is a good opportunity to onboard onto that way of thinking in the specs here because it's mostly just restructuring existing content.

Reply:
@smartcontracts I updated this PR and it is now split into three different documents:

- superchain-eth-bridge doc following the standardized format for contracts specs
- eth-liquidity docs both following the standardized format for contracts specs
- eth-bridging doc explaining the architecture of the ETH bridging system.

I also added SuperchainETHBridge and ETHLiquidity sections to the predeploys.md doc